pkg/odo/cli/component/create.go

```diff
@@ -585,8 +585,8 @@ func (co *CreateOptions) Validate() (err error) {
 			spinner := log.Spinner("Validating component")
 			defer spinner.End(false)
 
-			if util.CheckPathExists(DevfilePath) || util.CheckPathExists(EnvFilePath) {
-				return errors.New("Devfile.yaml or env.yaml already exists")
+			if util.CheckPathExists(EnvFilePath) {
```

Comment:
I still think if we remove this check we should handle the following two cases properly at the same time:
```
If devfile.yaml is present
Case 1: odo create <component type> -> Throw errors and tell user to delete the existing devfile.yaml then run odo create again
Case 2: odo create -> Use the existing devfile.yaml
```
Just simply remove this check can cause more issues such as for the current code, if user has nodejs devfile.yaml in the workspace directory but user runs `odo create maven`, it can still pass but that's not expected

Reply:
same as: https://github.com/openshift/odo/pull/2762#discussion_r401864713

Reply:
Actually... this is another new comment... not same as above test cases comments.... 🤦‍♂ It's about one use case this PR may break...and you should test that use case

Reply:
Actually I remember I should only handle Case2, and also the issue you created says that: https://github.com/openshift/odo/issues/2789
In addition, do think all the validation should be done within the same PR. 
If you want me to handle case1 listed in the PR, I don't mind added that in the PR. But I think it makes more sense to also cover `If devfile not exist` case as well. 

Please close your PR, and I will checkout your changes.

Reply:
Also the new use case test should be under `devfile-create` test. not the `devfile-push` or `devfile-url` I have modified in the PR.  And should be added within the same PR of devfile existence validation in `odo create`

Again, I don't mind work on that as well.

Reply:
This is my PR https://github.com/openshift/odo/pull/2798, if you want to take over that you can talk with @kadel to avoid confusion since I already asked him review and he may have more comments against the issue/PR

Reply:
Since you have already have all that implemented, I think it's better to keep it in your PR. 
You should also add the user case test in `devfile-create`

It makes more sense to leave this PR as it is, and we can modify the test code to match the expected behavior in no matter this PR or your PR. (As long as one of our PR merges, the test code would fail, we need to modify the test anyways.)